networkit/cpp/graph/GraphTools.cpp

```diff
@@ -74,6 +77,38 @@ Graph restoreGraph(std::vector<node>& invertedIdMap, const Graph& G) {
 	return Goriginal;
 }
 
+Graph extractLargestConnectedComponent(const Graph &G, bool compactGraph) {
+	Graph S(G);
+
+	if (!G.numberOfNodes())
+		return S;
+
+	ConnectedComponents cc(S);
+	cc.run();
+
+	auto compSizes = cc.getComponentSizes();
+	auto largestCC = std::max_element(
+		compSizes.begin(), compSizes.end(),
+		[](const std::pair<index, count> &x, const std::pair<index, count> &y) {
+			return x.second < y.second;
+		});
+	auto components = cc.getComponents();
+
+	for (index c = 0; c < components.size(); ++c) {
+		if (c == largestCC->first)
+			continue;
+		for (node u : components[c]) {
+			S.removeNode(u);
+		}
+	}
+
+	if (compactGraph) {
+		auto continuousNodeIds = getContinuousNodeIds(S);
+		S = getCompactedGraph(S, continuousNodeIds);
```

Comment:
If `compactGraph` is set, this method copies the Graph twice. We could probably get away with a single copy by deleting vertices and remapping the Graph at the same time.

Reply:
This can now be optimized using the getRemappedGraph() function from #324.